agent/functional_tests/tests/functionaltests_test.go

```diff
@@ -422,6 +423,63 @@ func TestCustomAttributesWithMaxOptions(t *testing.T) {
 	assert.True(t, ok, "OS attribute not found")
 }
 
+func waitForContainerHealthStatus(t *testing.T, testTask *TestTask) {
+	ctx, _ := context.WithTimeout(context.TODO(), waitTaskStateChangeDuration)
+waitHealth:
+	for {
+		select {
+		case <-ctx.Done():
+			t.Error("Waiting for container health status available timed out")
+		default:
+			testTask.Redescribe()
+			if aws.StringValue(testTask.Containers[0].HealthStatus) == "UNKNOWN" {
+				time.Sleep(time.Second)
+				continue
+			}
+			break waitHealth
+		}
+	}
+}
+
+func containerHealthWithoutStartPeriodTest(t *testing.T, taskDefinition string) {
+	RequireDockerVersion(t, ">=1.12.0")
+	RequireDockerVersion(t, "<17.05.0")
```

Comment:
Is this requirement necessary? Since the test can run without `startPeriod` even on greater versions

Reply:
yeah, but here I just want one of the tests to run. Either the test with `startperiod` should run or the test without `startperiod` should run. I don't see any benefit to run both of them.

Reply:
I think that's reasonable